### ISSUE TITLE
Better code documentation of the Gamma sampler

### ISSUE THREAD

**Issue by mschauer** [JuliaStats/Distributions.jl#835]
The Gamma sampler in https://github.com/JuliaStats/Distributions.jl/blob/master/src/samplers/gamma.jl will become used with #830, so it is desirable to document better how it is working.

**Comment by matbesancon** [comment]
@mschauer was this fixed by #830?

**Comment by mschauer** [comment]
No, actually the contrary: this issue was opened so 830 could proceed without having to worry about this for the moment.

**Comment by BMasinde** [comment]
Hi, is this still needed? If so I could take a look.
